.devcontainer/Dockerfile

```diff
@@ -35,61 +35,61 @@ RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
         nginx \
         # JSON tools
         jq \
-        jo
-
-# tedge dependencies
-# Install more recent version of mosquitto >= 2.0.18 from debian backports to avoid mosquitto following bugs:
-# The mosquitto repo can't be used as it does not included builds for arm64/aarch64 (only amd64 and armhf)
-# * https://github.com/eclipse/mosquitto/issues/2604 (2.0.11)
-# * https://github.com/eclipse/mosquitto/issues/2634 (2.0.15)
-RUN sh -c "echo 'deb [signed-by=/usr/share/keyrings/debian-archive-keyring.gpg] http://deb.debian.org/debian bookworm-backports main' > /etc/apt/sources.list.d/debian-bookworm-backports.list" \
+        jo \
+    # tedge dependencies
+    # Install more recent version of mosquitto >= 2.0.18 from debian backports to avoid mosquitto following bugs:
+    # The mosquitto repo can't be used as it does not included builds for arm64/aarch64 (only amd64 and armhf)
+    # * https://github.com/eclipse/mosquitto/issues/2604 (2.0.11)
+    # * https://github.com/eclipse/mosquitto/issues/2634 (2.0.15)
+    && sh -c "echo 'deb [signed-by=/usr/share/keyrings/debian-archive-keyring.gpg] http://deb.debian.org/debian bookworm-backports main' > /etc/apt/sources.list.d/debian-bookworm-backports.list" \
     && apt-get update \
     && DEBIAN_FRONTEND=noninteractive apt-get -y --no-install-recommends install -t bookworm-backports \
         mosquitto \
-        mosquitto-clients
-
-# Remove unnecessary systemd services
-RUN rm -f /lib/systemd/system/multi-user.target.wants/* \
-    /etc/systemd/system/*.wants/* \
-    /lib/systemd/system/local-fs.target.wants/* \
-    /lib/systemd/system/sockets.target.wants/*udev* \
-    /lib/systemd/system/sockets.target.wants/*initctl* \
-    /lib/systemd/system/systemd-update-utmp* \
+        mosquitto-clients \
+    #
+    # Remove unnecessary systemd services
+    && rm -f /lib/systemd/system/multi-user.target.wants/* \
+        /etc/systemd/system/*.wants/* \
+        /lib/systemd/system/local-fs.target.wants/* \
+        /lib/systemd/system/sockets.target.wants/*udev* \
+        /lib/systemd/system/sockets.target.wants/*initctl* \
+        /lib/systemd/system/systemd-update-utmp* \
+    #
     # Remove policy-rc.d file which prevents services from starting
-    && rm -f /usr/sbin/policy-rc.d
-
-# Install gh utility
-RUN curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg \
+    && rm -f /usr/sbin/policy-rc.d \
+    #
+    # Install gh utility
+    && curl -fsSL https://cli.github.com/packages/githubcli-archive-keyring.gpg | sudo dd of=/usr/share/keyrings/githubcli-archive-keyring.gpg \
```

Comment:
What is the benefit of chaining of all these commands one after the other, as opposed to individual `RUN` statements? To avoid caching all the intermediary layers?

Wouldn't it make the debugging harder when any one of those steps fail, as we'd struggle to narrow-down which command in the chain failed?

Reply:
Yes, using a single run reduces the number of layers (`RUN` creates a new layer). Debugging still works as you still get an error when something is wrong, so it is not that difficult (all the other public docker images do it anyway ;)).